src/Compilers/Test/Core/MarkedSource/SourceWithMarkedNodes.cs

```diff
@@ -8,41 +8,48 @@
 using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.Linq;
-using System.Text;
 using System.Text.RegularExpressions;
-using System.Threading.Tasks;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Text;
-using Roslyn.Utilities;
 using Xunit;
 
 namespace Roslyn.Test.Utilities
 {
-    internal sealed class SourceWithMarkedNodes
+    internal sealed partial class SourceWithMarkedNodes
     {
         public readonly string Source;
+        public readonly string StrippedSource;
+        public readonly string Input;
         public readonly SyntaxTree Tree;
+        public readonly ImmutableArray<MarkedSpan> MarkedSpans;
         public readonly ImmutableArray<ValueTuple<TextSpan, int, int>> SpansAndKindsAndIds;
 
         public SourceWithMarkedNodes(string markedSource, Func<string, SyntaxTree> parser, Func<string, int> getSyntaxKind)
         {
             Source = ClearTags(markedSource);
+            StrippedSource = RemoveTags(markedSource);
```

Comment:
Perhaps it would be better to pass in a flag that determines whether we clear the tags or remove them, instead of doing both.

Reply:
I was going to do that, but it felt like a bit of a trap. If we remove the tags then all of the offsets are wrong for all but the first tag, so I though "You can use `StrippedSource` if you know what your doing" was better than "if you pass in `removeTags: true` you might end up surprised".

I'm not too fussed if you think thats not a problem though

Reply:
I'm not sure if there is much difference in the usage - either way you need to explicitly opt-into removing the tags. Whether it's via a parameter or a different property doesn't seem to make difference.